tests/hash.c

```diff
@@ -139,30 +139,34 @@ int HashTest(void)
     #endif
 
     #ifndef NO_SHA
-    if ( (ret = hmac_sha_test()) )
+    if ( (ret = hmac_sha_test()) ) {
         printf( "   HMAC-SHA test failed!\n");
-    else
+        return ret;
```

Comment:
Why are we terminating the remainder of tests now? That was not the previous behavior.

Reply:
Ah thanks, I meant to point that out to you to make sure I'd guessed right.  The 9 preceding tests all `return ret` on failure, and I was guessing it was accidental that the 4 final tests were discarding their `ret`s and continuing, rather than `return`ing like the other 9.  Obviously we want to expose the `ret` to the caller, and the code as-was did that for the first 9 but not for the final 4.